docs/adr/adr-009-public-api.md

```diff
@@ -0,0 +1,438 @@
+// Multiple Subscriptions can be created.
+Subscribe() (Subscription, error)
+// SyncState returns the current state of the header Syncer. 
+SyncState() sync.State
+// SyncWait blocks until the header Syncer is synced to network head. 
+SyncWait(ctx context.Context) error
+// SyncHead provides the Syncer's view of the current network head.
+SyncHead(ctx context.Context) (*header.ExtendedHeader, error)
+}
+```
+
+##### Shares
+
+```go
+  type SharesModule interface {
+    // GetShare returns the Share from the given data Root at the given row/col
+    // coordinates.
+    GetShare(ctx context.Context, root *Root, row, col int) (Share, error)
+    // GetSharesByNamespace returns all shares of the given nID from the given data
+    // Root.
+    GetSharesByNamespace(
+      ctx context.Context,
+      root *Root, 
+      nID namespace.ID, 
+    ) ([]Share, error)
+    // SharesAvailable subjectively validates if Shares committed to the given data
+    // Root are available on the network. 
+    SharesAvailable(ctx context.Context, root *Root) error
+    // ProbabilityOfAvailability calculates the probability of the data square
+    // being available based on the number of samples collected.
+    ProbabilityOfAvailability() float64
+  }
+```
+
+##### P2P
+
+```go
+  type P2PModule interface {
+    // Info returns basic information about the node's p2p host/operations.
+    Info() p2p.Info
+    // Peers returns all peer IDs used across all inner stores.
+    Peers() peer.IDSlice 
+    // PeerInfo returns a small slice of information Peerstore has on the
+    // given peer.
+    PeerInfo(id peer.ID) peer.AddrInfo
+   
+    // Connect ensures there is a connection between this host and the peer with
+    // given peer.
+    Connect(ctx context.Context, pi peer.AddrInfo) error
+    // ClosePeer closes the connection to a given peer. 
+    ClosePeer(id peer.ID) error
+    // Connectedness returns a state signaling connection capabilities.
+    Connectedness(id peer.ID) network.Connectedness
+    // NATStatus returns the current NAT status.
+    NATStatus() network.Reachability
+   
+    // BlockPeer adds a peer to the set of blocked peers.
+    BlockPeer(p peer.ID) error
+    // UnblockPeer removes a peer from the set of blocked peers.
+    UnblockPeer(p peer.ID) error
+    // ListBlockedPeers returns a list of blocked peers.
+    ListBlockedPeers() []peer.ID
+
+    // MutualAdd adds a peer to the list of peers who have a bidirectional
+    // peering agreement that they are protected from being trimmed, dropped
+    // or negatively scored.
+    MutualAdd(id peer.ID, tag string)
+    // MutualAdd removes a peer from the list of peers who have a bidirectional
+    // peering agreement that they are protected from being trimmed, dropped
+    // or negatively scored, returning a bool representing whether the given
+    // peer is protected or not.
+    MutualRm(id peer.ID, tag string) bool
```

Comment:
This is confusing: If I want to remove a peer and `MutualRm` returns `true`,  it means the peer was protected? Does `true` mean it does not get removed then?